backend/mechanismbackend/api/models.py

```diff
@@ -47,3 +47,33 @@ def clean(self):
                 not self.outputT3):
             raise ValidationError("Mechanism needs at least one input or output to be specified.")
 
+    def get_dissimilarity(self, other):
+        max_shared_attrs = 12 + 1
+        shared_attrs = 0
+
+        if self.inputR1 == other.inputR1:
+            shared_attrs += 1
+        if self.inputR2 == other.inputR2:
+            shared_attrs += 1
+        if self.inputR3 == other.inputR3:
+            shared_attrs += 1
+        if self.inputT1 == other.inputT1:
+            shared_attrs += 1
+        if self.inputT2 == other.inputT2:
+            shared_attrs += 1
+        if self.inputT3 == other.inputT3:
+            shared_attrs += 1
+        if self.outputR1 == other.outputR1:
+            shared_attrs += 1
+        if self.outputR2 == other.outputR2:
+            shared_attrs += 1
+        if self.outputR3 == other.outputR3:
+            shared_attrs += 1
+        if self.outputT1 == other.outputT1:
+            shared_attrs += 1
+        if self.outputT2 == other.outputT2:
+            shared_attrs += 1
+        if self.outputT3 == other.outputT3:
+            shared_attrs += 1
+
+        return 1 / (shared_attrs / max_shared_attrs + 0.01)
```

Comment:
Please also add a metric to compare the transmissions and names of the mechanisms. For the names, you could use, e.g, the Levenshtein distance.

Reply:
As long as transmission is an abstract number I do not think we should incorporate it.
For names I agree and plan to implement it

Reply:
We could at least check whether the transmissions do perfectly match (as long as they are not both undefined).